usgs-details.html

```diff
@@ -53,46 +106,57 @@
         }
 
         // Function to capture results of USGS and populate values
-        function listener() {
-            if (this.status === 200) {
-                var json = JSON.parse(this.responseText);
-
-                try {
-                    var value = json.value.timeSeries[0].values[0].value[0].value;
-                    var units = json.value.timeSeries[0].variable.unit.unitCode;
-                    console.log(value + ' ' + units);
-
-                    document.getElementById('value').innerText = value;
-                    document.getElementById('units').innerText = units;
-                } catch (ex) {
-                    console.warn('Error while parsing result: ' + ex.message);
+        function listenerFor(metric) {
+            return function() {
+                var value = metric.fallback[new Date().getMonth()];
+                var units = metric.units;
+
+                if (this.status === 200) {
+                    var json = JSON.parse(this.responseText);
+
+                    try {
+                        var apiValue = json.value.timeSeries[0].values[0].value[0].value;
+                        var apiUnits = json.value.timeSeries[0].variable.unit.unitCode;
+                        console.log('From API: ' + apiValue + ' ' + apiUnits);
+
+                        value = metric.transform(apiValue);
+                    } catch (ex) {
+                        console.warn('Error while parsing result: ' + ex.message);
+                    }
+                } else {
+                    console.warn('Error while fetching USGS data: ' + this.status);
+                    console.log('Using fallback values');
                 }
-            } else {
-                console.warn('Error while fetching USGS data: ' + this.status);
-            }
-        }
 
-        // Function to fetch values from USGS given a code
-        function fetchDataForCode(code) {
-            if (!code) {
-                console.warn('Empty code');
-                return;
-            }
+                document.getElementById('value').innerText = value;
+                document.getElementById('units').innerText = units;
+            };
+        }
 
-            var url = REQUEST_URL + '&parameterCd=' + code;
+        // Function to fetch values from USGS given a metric
+        function fetchDataForMetric(metric) {
+            var url = REQUEST_URL + '&parameterCd=' + metric.code;
```

Comment:
Consider writing a `createRequestUrl(code)` function to encapsulate this as well as the `BASE_URL` and `REQUEST_URL` var.